File: Android/shared/src/main/kotlin/shared/dto/primitives/Address.kt
package shared.dto.primitives

import kotlinx.serialization.Serializable
import shared.dto.protocols.Parametable
import shared.dto.protocols.Responsable

@Serializable
data class Address(
    val address: String,
    val city: String,
    val country: String,
    val latitude: Double,
    val longitude: Double,
) : Parametable(), Responsable

File: Node/src/dto/primitives/Address.ts
import { Max, Min } from "class-validator"

export class Address {
    address: string
    city: string
    country: string

    @Max(90)
    @Min(-90)
    latitude: number

    @Max(180)
    @Min(-180)
    longitude: number

    constructor(
        address: string,
        city: string,
        country: string,
        latitude: number,
        longitude: number
    ) {
        this.address = address
        this.city = city
        this.country = country
        this.latitude = latitude
        this.longitude = longitude
    }
}
